### PR TITLE
Enable editing maxCharLength for some text fields

### LINKED ISSUE #3468: Increase "Next Steps" characters from 250 characters to 1,000 characters
**Is your feature request related to a problem? Please describe.**
Users are complaining that there isn't enough space for writing the Next Steps in an engagement report.

**Describe the solution you'd like**
Change the number of characters for "Next Steps" from 250 to 1,000

**Describe alternatives you've considered**
N/A

**Additional context**
This request has been approved by the requirements owner

The nextSteps column in the database is nvarchar(max)

### PR DESCRIPTION
**Pull request by cemalettin-simsoft**
Provide default first, add any exceptions as additional properties in dictionary.yml

Closes #3468

#### User changes
- Users will be able to write longer texts for text fields (if the dictionary for that field allows it).

#### Super User changes
- none

#### Admin changes
- none

#### System admin changes
- [X] anet.yml or anet-dictionary.yml needs change
  Previously `fields.report.nextSteps` was just a text like this in the dictionary:
  ```
    nextSteps: Next steps
  ```
  Now it is a nested field:
  ```
    nextSteps: 
      label: Next steps
      maxTextFieldLength: 1000
  ````
- [ ] db needs migration
- [X] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [X] Described the user behavior in PR body
  - [X] Referenced/updated all related issues
  - [X] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [X] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [ ] Added and/or updated unit tests
  - [ ] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [X] Updated documentation
  - [X] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: client/src/pages/admin/authorizationgroup/Form.js [hunk 1]
```diff
@@ -104,18 +104,24 @@ const AuthorizationGroupForm = ({ edit, title, initialValues }) => {
                   name="description"
                   component={FieldHelper.InputField}
                   componentClass="textarea"
-                  maxLength={Settings.maxTextFieldLength}
+                  maxLength={
+                    Settings.maxTextFieldLength?.authorizationGroup
+                      ?.description || Settings.maxTextFieldLength.default
```
Comment (gjvoosten) [inline client/src/pages/admin/authorizationgroup/Form.js:109]: Too DRY (same expression is repeated three times); extract into a const, and perhaps provide a util method `getMaxTextFieldLength(setting)` that when called as
`getMaxTextFieldLength(Settings.maxTextFieldLength?.authorizationGroup?.description)` will either return the setting from the dictionary if defined, or the default.

File: docs/INSTALL.md [hunk 2]
```diff
@@ -192,9 +192,17 @@ dictionary:
...
+    default: 250
+    report:
+      nextSteps: 1000
+      intent: 400
+      keyOutcomes: 600
+    authorizationGroup:
+      description: 400
 
-  fields:
+fields:
```
Comment (gjvoosten) [inline docs/INSTALL.md:205]: If you want to remove the spaces here, remove the top-level `dictionary:` key (as that no longer exists) and shift everything left by two spaces (docs are incorrect currently).

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: anet-dictionary.yml [hunk 3]
```diff
@@ -27,7 +27,15 @@ reportWorkflow:
   nbOfHoursQuarantineApproved: 24
   nbOfHoursApprovalTimeout: 48
 
-maxTextFieldLength: 250
+# Add exceptions if there is any, only `default` is mandatory, others are optional
+maxTextFieldLength:
+  default: 250
+  report:
+    nextSteps: 1000
```
Comment (VassilIordanov) [inline anet-dictionary.yml:34]: I suggest defining this value in the definition of the field instead of a central place for text field lenths

Comment (cemalettin-simsoft) [reply anet-dictionary.yml:34]: And for the default value, we use the older version?

Comment (cemalettin-simsoft) [reply anet-dictionary.yml:34]: This also requires us to add additional fields to dictionary, for `authorizationGroups` for example, there are no fields, we will need to enable those in the schema as well. That is a lot of change for a simple issue

**Review by gjvoosten — APPROVED** [review]

**Review by gjvoosten — APPROVED** [review]

**Pull request merged**